### ISSUE TITLE
View assignment 1D mismatched layout with static extents fails to compile

### ISSUE THREAD

**Issue by crtrott** [kokkos/kokkos#5534]
```
#include <Kokkos_Core.hpp>
#include <cmath>

int main(int argc, char* argv[]) {
  Kokkos::initialize(argc, argv);
  {
    int N = argc > 1 ? atoi(argv[1]) : 1000000;
    int R = argc > 2 ? atoi(argv[2]) : 10;

    Kokkos::View<double[5],Kokkos::LayoutLeft> a("A");
    Kokkos::View<double[5],Kokkos::LayoutRight> b("B");
    a = b;
    b = a;

  }
  Kokkos::finalize();
}
```
```
/ascldap/users/crtrott/Kokkos/kokkos/core/src/Kokkos_View.hpp:1351:28: error: static assertion failed: Incompatible View copy assignment
 1351 |     static_assert(Mapping::is_assignable, "Incompatible View copy assignment");
```

**Comment by crtrott** [comment]
Failed forever ... (at least back to 3.0)

**Comment by crtrott** [comment]
Hoping this is easy to fix .. @lucbv ran into this.

**Comment by PhilMiller** [comment]
I can totally see why that assignment should be workable, but is it something that actually should be allowed?

Is there an implicit statement in this report that the corresponding case with dynamic extents does compile, and works as expected, and is the behavior we want to expand?

**Comment by crtrott** [comment]
yeah I found the places already. And yes this should work, mdspan does allow it.

**Comment by PhilMiller** [comment]
Ok, I guess I shouldn't go off and derail mdspan standardization by requestioning this :-D

**Comment by crtrott** [comment]
It also works already for matching layouts in Kokkos View.

**Comment by crtrott** [comment]
And the actual part in the code which is responsible for checking extents compatibility already did the right thing.

**Comment by PhilMiller** [comment]
Given that this is long standing behavior, shipping a change in a patch release is nevertheless iffy, even if it's safe

**Comment by crtrott** [comment]
yeah but it only enables stuff which didn't compile before, i.e. very unlikely to break any code. Note that the error is a static assert not an SFINAE can't find the overload, so it shouldn't mess with overload sets.

**Comment by PhilMiller** [comment]
Fixed in 3.7.01 with #5566
